src/Analyzers/CSharp/CodeFixes/RemoveUnusedParametersAndValues/CSharpRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -94,15 +95,28 @@ protected override SyntaxNode TryUpdateNameForFlaggedNode(SyntaxNode node, Synta
 
         protected override SyntaxNode TryUpdateParentOfUpdatedNode(SyntaxNode parent, SyntaxNode newNameNode, SyntaxEditor editor, ISyntaxFacts syntaxFacts, SemanticModel semanticModel)
         {
-            if (newNameNode.IsKind(SyntaxKind.DiscardDesignation)
-                && parent is DeclarationPatternSyntax declarationPattern
-                && parent.SyntaxTree.Options.LanguageVersion() >= LanguageVersion.CSharp9)
+            if (newNameNode.IsKind(SyntaxKind.DiscardDesignation))
             {
-                var trailingTrivia = declarationPattern.Type.GetTrailingTrivia()
-                    .AddRange(newNameNode.GetLeadingTrivia())
-                    .AddRange(newNameNode.GetTrailingTrivia());
+                var triviaToAppend = newNameNode.GetLeadingTrivia().AddRange(newNameNode.GetTrailingTrivia());
 
-                return SyntaxFactory.TypePattern(declarationPattern.Type).WithTrailingTrivia(trailingTrivia);
+                if (parent is DeclarationPatternSyntax declarationPattern &&
+                    parent.SyntaxTree.Options.LanguageVersion() >= LanguageVersion.CSharp9)
```

Comment:
do you have a test for the C#8 case?

Reply:
The case for declaration patterns was here before, so I guess it is already properly tested. The change to this part is due to extracting of `triviaToAppend` into a separate local since it is used in all 3 cases.